src/test/osd/RadosModel.h

```diff
@@ -2349,39 +2513,31 @@ class SetChunkOp : public TestOp {
       if ((r = comp->get_return_value())) {
 	if (r == -ENOENT && src_value.deleted()) {
 	  cout << num << ":  got expected ENOENT (src dne)" << std::endl;
+	} else if (r == -ENOENT && context->oid_set_chunk_tgt_pool.find(oid_tgt) != 
+		  context->oid_set_chunk_tgt_pool.end()) {
+	  cout << num << ": get expected ENOENT tgt oid " << oid_tgt << std::endl;
+	} else if (r == -ERANGE && src_value.deleted()) {
+	  cout << num << ":  got expected ERANGE (src dne)" << std::endl;
 	} else if (r == -EOPNOTSUPP) {
-	  bool is_overlapped = false;
-	  for (auto &p : src_value.chunk_info) {
-	    if ((p.first <= offset && p.first + p.second.length > offset) ||
-                (p.first > offset && p.first <= offset + length)) {
-	      cout << " range is overlapped  offset: " << offset << " length: " << length
-		    << " chunk_info offset: " << p.second.offset << " length " 
-		    << p.second.length << std::endl;
-	      is_overlapped = true;
-	      context->update_object_version(oid, comp->get_version64());
-	    } 
-	  }
-	  if (!is_overlapped) {
-	    cerr << "Error: oid " << oid << " set_chunk " << oid_tgt << " returned error code "
-		  << r << " offset: " << offset << " length: " << length <<  std::endl;
-	    ceph_abort();
-	  }
+	  cout << "Range is overlapped: oid " << oid << " set_chunk " << oid_tgt << " returned error code "
+		<< r << " offset: " << offset << " length: " << length <<  std::endl;
+	  context->update_object_version(oid, comp->get_version64());
 	} else {
 	  cerr << "Error: oid " << oid << " set_chunk " << oid_tgt << " returned error code "
 	       << r << std::endl;
 	  ceph_abort();
 	}
       } else {
-	ChunkDesc info;
-	info.offset = tgt_offset;
-	info.length = length;
-	info.oid = oid_tgt;
-	context->update_object_chunk_target(oid, offset, info);
-	context->update_object_version(oid, comp->get_version64());
+	if (snap == -1) {
+	  ChunkDesc info {tgt_offset, length, oid_tgt};
+	  context->update_object_chunk_target(oid, offset, info);
+	  context->update_object_version(oid, comp->get_version64());
```

Comment:
Is this actually only valid for HEAD?  Wouldn't we also need to update chunk_target for a clone if we call SET_CHUNK on a clone?

Reply:
Related: how does chunk_target relate to FLUSH?  We don't appear to populate it in that case.

Reply:
ChunkDesc is only for SetChunkReadOp to check whether chunk entries in chunk_map can be read correctly. Also, SetChunkReadOp is only used in case the object is head (/qa/suites/rados/thrash/workloads/set-chunks-read.yaml).
Probably, deleting SetCunkReadOp is the right way to address your comment?